futurex_openedx_extensions/dashboard/views.py

```diff
@@ -1223,33 +1228,73 @@ class ThemeConfigDraftView(FXViewRoleInfoMixin, APIView):
     permission_classes = [FXHasTenantCourseAccess]
     fx_view_name = 'theme_config_draft'
     fx_allowed_write_methods = ['PUT', 'DELETE']
-    fx_view_description = 'api/fx/config/v1/draft/: draft theme config APIs'
+    fx_view_description = 'api/fx/config/v1/draft/<tenant_id>: draft theme config APIs'
 
-    def get(self, request: Any) -> Response | JsonResponse:  # pylint: disable=no-self-use
+    def get(self, request: Any, tenant_id: int) -> Response | JsonResponse:  # pylint: disable=no-self-use
         """Get draft config"""
-
+        updated_fields = get_draft_tenant_config(int(tenant_id))
+        dict_str = json.dumps(updated_fields, sort_keys=True, separators=(',', ':')) if updated_fields else ''
         return JsonResponse({
-            'updated_fields': {
-                'platform_name': {
-                    'published_value': 'my platform name',
-                    'draft_value': 'My Awesome Platform'
-                },
-                'primary_color': {
-                    'published_value': '#ff0000',
-                    'draft_value': '#ffff00'
-                },
-            },
-            'draft_hash': 'ajsd90a8su9a8u9a8sdyf0a9sdhy0asdjgasdgkjdsfgj'
+            'updated_fields': updated_fields,
+            'draft_hash': hashlib.sha256(dict_str.encode()).hexdigest() if dict_str else ''
         })
 
-    def put(self, request: Any) -> Response:  # pylint: disable=no-self-use
+    def put(self, request: Any, tenant_id: int) -> Response:  # pylint: disable=no-self-use
         """Update draft config"""
+        data = request.data
 
-        return Response(status=http_status.HTTP_204_NO_CONTENT)
+        try:
+            if not isinstance(data['key'], str):
+                raise FXCodedException(
+                    code=FXExceptionCodes.INVALID_INPUT, message='Key name must be a string.'
+                )
 
-    def delete(self, request: Any) -> Response:  # pylint: disable=no-self-use
-        """Delete draft config"""
+            key_access_info = ConfigAccessControl.objects.get(key_name=data['key'])
+            if not key_access_info.writable:
+                raise FXCodedException(
+                    code=FXExceptionCodes.INVALID_INPUT, message=f'Config Key: ({data["key"]}) is not writable.'
+                )
+
+            new_value = data.get('new_value')
+            reset = data.get('reset', False)
+            if not reset and not new_value:
+                raise FXCodedException(
+                    code=FXExceptionCodes.INVALID_INPUT, message='Provide either new_value or reset.'
+                )
+            if new_value and not isinstance(new_value, type(data['current_value'])):
+                raise FXCodedException(
+                    code=FXExceptionCodes.INVALID_INPUT, message='New value type must match current value.'
+                )
```

Comment:
these validations are great, but here we have a hidden bug. `new_value` might exist but have `None` as a value. Same for `current_value`; it can be `None` (which will cause an exception every time we call the API for that key)

I'll move the discussion to slack, then we post here our discussion: will we allow `None` for config values or not